### PR TITLE
Internal: Create the user completes all steps step [ED-15448]

### PR DESCRIPTION
**Pull request by elchugreeva**

### CODE REVIEW

File: modules/checklist/assets/js/app/components/success-message.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	};
+
+	return (
+		<Card elevation={ 0 } square={ true } className="all_done">
+			<CardMedia
+				image="https://assets.elementor.com/checklist/v1/images/checklist-step-7.jpg"
+				sx={ { height: 180 } }
+			/>
+			<CardContent sx={ { textAlign: 'center' } }>
+				<Typography variant="h6" color="text.primary">You&apos;re on your way!</Typography>
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/success-message.js:23]:
```suggestion
			<CardContent sx={ { textAlign: 'center' } }>
				<Typography variant="h6" color="text.primary">{ __( 'You&apos;re on your way!' ) }</Typography>
```

File: modules/checklist/assets/js/app/components/success-message.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+				<Typography variant="body2" color="text.secondary" component="p">
+					With these steps, you&apos;ve got a great base for a robust website. Enjoy your web creation journey!
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/success-message.js:25]: Same here

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 3]
```diff
@@ -243,4 +243,90 @@ test.describe( 'Launchpad checklist tests', () => {
...
+			rocketButton = editor.page.locator( selectors.topBarIcon ),
+			closeButton = editor.page.locator( selectors.closeButton ),
+			checklist = editor.page.locator( selectors.popup ),
+			allDone = editor.page.locator( selectors.allDone ),
+			gotItButton = editor.page.locator( selectors.gotItButton ),
+			url = '/wp-json/elementor/v1/checklist/steps';
+
+		const checklistHelper = new ChecklistHelper( page, testInfo, apiRequests );
+
+		const returnDataMock = ( isCompleted ) => {
```
Comment (ronenelementor) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:259]: This is awesome!
Please put it in the helper

File: modules/checklist/assets/js/app/components/success-message.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+				show_launchpad_checklist: '',
+			},
+			options: {
+				external: true,
+			},
+		} );
+	};
+
+	return (
+		<Card elevation={ 0 } square={ true } className="all_done">
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/success-message.js:17]: We were asked not to add classnames just for testing purposes unfortunately

File: modules/checklist/assets/js/app/components/checklist-wrapper.js [hunk 5]
```diff
@@ -2,27 +2,33 @@ import { useState } from 'react';
...
+								setExpandedIndex={ setExpandedIndex }
+								expandedIndex={ expandedIndex }
+								index={ index }
+							/>
+						);
+					} )
+				}
+			</List>
+			{ checkIfAllStepsCompleted ? <SuccessMessage /> : null }
+		</>
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/checklist-wrapper.js:31]: I don't think that both the step list and the success message should be displayed at the same time
```suggestion
		checkIfAllStepsCompleted
			? <SuccessMessage />
			:  <List component="div" sx={ { py: 0 } }>
				{
					steps.map( ( step, index ) => {
						return (
							<CheckListItem
								key={ index }
								step={ step }
								setSteps={ setSteps }
								setExpandedIndex={ setExpandedIndex }
								expandedIndex={ expandedIndex }
								index={ index }
							/>
						);
					} )
				}
			</List>
```

File: tests/playwright/sanity/modules/checklist/selectors.ts [hunk 6]
```diff
@@ -20,6 +20,8 @@ const selectors = {
 	progressBarPercentage: '#e-checklist header > div:nth-child(2) > div:nth-child(2)',
 	infotipFirstTimeClosed: '.e-checklist-infotip-first-time-closed',
 	infotipFirstTimeClosedButton: '.infotip-first-time-closed-button',
+	allDone: '.all_done',
+	gotItButton: '.all_done button',
```
Comment (ronenelementor) [inline tests/playwright/sanity/modules/checklist/selectors.ts:24]:
```suggestion
	allDone: '#e-checklist . MuiPaper-root . MuiPaper-root',
	gotItButton: '#e-checklist . MuiPaper-root . MuiPaper-root button',
```

File: modules/checklist/assets/js/app/components/checklist-wrapper.js [hunk 7]
```diff
@@ -2,27 +2,33 @@ import { useState } from 'react';
 import { List } from '@elementor/ui';
 import CheckListItem from './checklist-item';
 import PropTypes from 'prop-types';
+import SuccessMessage from './success-message';
+import { isStepChecked } from '../../utils/functions';
 
 const ChecklistWrapper = ( { steps, setSteps } ) => {
 	const [ expandedIndex, setExpandedIndex ] = useState( -1 );
+	const checkIfAllStepsCompleted = steps.filter( isStepChecked ).length === steps.length;
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/checklist-wrapper.js:10]: `Check if` implies a function
```suggestion
	const isChecklistCompleted = steps.filter( isStepChecked ).length === steps.length;
```

File: modules/checklist/assets/js/app/components/success-message.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+		$e.run( 'document/elements/settings', {
+			container: elementor.settings.editorPreferences.getEditedView().getContainer(),
+			settings: {
+				show_launchpad_checklist: '',
+			},
+			options: {
+				external: true,
+			},
+		} );
+	};
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/success-message.js:14]: This is too good to be here IMO
I would move it to the utils/functions.js to be used here and later in more places

File: modules/checklist/assets/js/app/components/success-message.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+	return (
+		<Card elevation={ 0 } square={ true } className="all_done">
+			<CardMedia
+				image="https://assets.elementor.com/checklist/v1/images/checklist-step-7.jpg"
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/success-message.js:19]: @louiswol94 might know if we need to sanitize this, and how to

File: tests/playwright/sanity/modules/checklist/selectors.ts (hunk 6 above)
Comment (elchugreeva) [reply tests/playwright/sanity/modules/checklist/selectors.ts:24]: I think in this specific case it is justified, @ronenelementor @louiswol94 WDYT?

File: modules/checklist/assets/js/app/components/success-message.js (hunk 2 above)
Comment (elchugreeva) [reply modules/checklist/assets/js/app/components/success-message.js:25]: done

File: modules/checklist/assets/js/app/components/checklist-wrapper.js (hunk 7 above)
Comment (elchugreeva) [reply modules/checklist/assets/js/app/components/checklist-wrapper.js:10]: done

File: modules/checklist/assets/js/app/components/success-message.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+const SuccessMessage = () => {
+	return (
+		<Card elevation={ 0 } square={ true } className="all_done">
+			<CardMedia
+				image="https://assets.elementor.com/checklist/v1/images/checklist-step-7.jpg"
+				sx={ { height: 180 } }
+			/>
+			<CardContent sx={ { textAlign: 'center' } }>
+				<Typography variant="h6" color="text.primary">{ __( 'You\'re on your way!' ) }</Typography>
```
Comment (louiswol94) [inline modules/checklist/assets/js/app/components/success-message.js:12]:
```suggestion
				<Typography variant="h6" color="text.primary">{ __( 'You\'re on your way!', 'elementor' ) }</Typography>
```

File: modules/checklist/assets/js/app/components/success-message.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+				</Typography>
+			</CardContent>
+			<CardActions sx={ { justifyContent: 'center' } }>
+				<Button
+					color="primary"
+					size="small"
+					variant="contained"
+					onClick={ hideChecklist }
+				>
+					{ __( 'Got it' ) }
```
Comment (louiswol94) [inline modules/checklist/assets/js/app/components/success-message.js:24]:
```suggestion
					{ __( 'Got it', 'elementor' ) }
```
And others

Comment (elchugreeva) [reply modules/checklist/assets/js/app/components/success-message.js:24]: fixed

File: modules/checklist/assets/js/app/components/success-message.js (hunk 9 above)
Comment (louiswol94) [reply modules/checklist/assets/js/app/components/success-message.js:8]: It's ok, it's hard coded :)

File: modules/checklist/assets/js/utils/functions.js [hunk 12]
```diff
@@ -6,6 +6,18 @@ export function isStepChecked( step ) {
 	return ! step[ IS_LOCKED ] && ( step[ IS_MARKED_COMPLETED ] || step[ IS_ABSOLUTE_COMPLETED ] || step[ IS_IMMUTABLE_COMPLETED ] );
 }
 
+export function hideChecklist() {
+	$e.run( 'document/elements/settings', {
+		container: elementor.settings.editorPreferences.getEditedView().getContainer(),
+		settings: {
+			show_launchpad_checklist: '',
```
Comment (louiswol94) [inline modules/checklist/assets/js/utils/functions.js:13]: Why is this a setting on a specific container?

File: modules/checklist/assets/js/app/components/success-message.js (hunk 9 above)
Comment (elchugreeva) [reply modules/checklist/assets/js/app/components/success-message.js:8]: perfect

File: tests/playwright/sanity/modules/checklist/selectors.ts (hunk 6 above)
Comment (louiswol94) [reply tests/playwright/sanity/modules/checklist/selectors.ts:24]: I think it's ok. Maybe just use something like `e-checklist-done`

**Review by ronenboxer — DISMISSED** [review]

File: modules/checklist/assets/js/app/components/success-message.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+			},
+		} );
+	};
+
+	useEffect( () => {
+		switch ( ajaxState.status ) {
+			case 'success':
+				setTimeout( () => {
+					$e.commands.run( 'checklist/toggle-icon', false );
+					}, 0 );
```
Comment (louiswol94) [inline modules/checklist/assets/js/app/components/success-message.js:31]:
```suggestion
			case 'success':
				setTimeout( () => {
					$e.commands.run( 'checklist/toggle-icon', false );
				}, 0 );
```

**Review by louiswol94 — APPROVED** [review]

**Review by ronenboxer — APPROVED** [review]

**Pull request merged**